R/build-search-docs.R

```diff
@@ -65,3 +68,138 @@ url_node <- function(url) {
     paste0("<url><loc>", url, "</loc></url>")
   )
 }
+
+#' Build search index
+#'
+#' Build a JSON file encompassing all HTML pages, for use by the search script.
+#'
+#' @section YAML config:
+#' You can exclude some paths from the search index.
+#' Below we exclude the changelog from the search index.
+#'
+#' ```yaml
+#' search:
+#'   exclude: ['news/index.html']
+#' ```
+#' @section Local testing:
+#' Search won't work if you simply use pkgdown preview of the static files.
+#' You can use `servr::httw("docs")` instead.
+#'
+#'
+#' @inheritParams build_articles
+#' @export
+#'
+build_search <- function(pkg = ".",
+                         override = list()) {
+  pkg <- section_init(pkg, depth = 1L, override = override)
+  rule("Building search index")
+
+  paths <- fs::path_rel(
+      fs::dir_ls(pkg$dst_path, glob = "*.html", recurse = TRUE),
+      pkg$dst_path
+  )
+  paths <- paths[!paths %in% c("404.html", "articles/index.html", "reference/index.html")]
```

Comment:
No need to change this code, but `setdiff()` is a nice option for this sort of problem.